app/script/message/MentionEntity.js

```diff
@@ -22,31 +22,57 @@
 window.z = window.z || {};
 window.z.message = z.message || {};
 
-z.message.Mention = class Mention {
-  constructor(start, end, userId = '') {
-    this.end = end;
-    this.start = start;
+z.message.MentionEntity = class Mention {
+  constructor(mention = {}) {
+    this.startIndex = mention.start || 0;
+    this.length = mention.length || 1;
+    this.type = mention.mention_type || z.cryptography.PROTO_MESSAGE_TYPE.MENTION_TYPE_USER_ID;
 
-    this.userId = userId;
+    this.userId = mention.user_id || '';
+  }
+
+  get endIndex() {
+    return this.startIndex + this.length - 1;
+  }
+
+  isValid(messageText = '') {
```

Comment:
Some tests for this method would be nice (non-blocking, could be done in another PR)

Reply:
Fair point. I will try to look into that later today. Would want to merge this now though so that @AndyLnd can continue his work on the rich input bar.

Reply:
:+1: